server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -117,12 +119,28 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
                 throw new IllegalArgumentException(unrecognized(request, unconsumedParams, candidateParams, "parameter"));
             }
 
-            if (request.hasContent() && request.isContentConsumed() == false) {
+            if (request.hasContent() && (request.isContentConsumed() == false && request.isFullContent())) {
                 throw new IllegalArgumentException(
                     "request [" + request.method() + " " + request.path() + "] does not support having a body"
                 );
             }
 
+            if (request.isStreamedContent()) {
+                assert action instanceof RequestBodyChunkConsumer;
+                var chunkConsumer = (RequestBodyChunkConsumer) action;
```

Comment:
I think this is where we'd fail if there is a discrepancy between the handlers and the `decider` filter in `Netty4HttpAggregator`. This is fine for now (for the merge), and possibly also for our initial roll-out plans.

I do wonder though if we could have a simple aggregating handler here that protects us against a discrepancy, in particular before we enable this for regular releases (or perhaps before that even)? Would it be feasible to always do the aggregation here even (when the mode is enabled), assuming larger requests are handled by proper handlers?

Reply:
It's indeed a temporary measure to do minimal change in http/rest layer to allow partial content. Current intention is to remove HttpAggregator and do aggregation in RestHandler(Base?). I drafted it before in #112120 and planning to revisit once we merge to main. There are several options how to do that, I will prepare a design doc.